base/database/database.go

```diff
@@ -13,11 +12,14 @@ func OnConflictUpdate(db *gorm.DB, key string, updateCols ...string) *gorm.DB {
 
 // Appends `ON CONFLICT (key...) DO UPDATE SET (fields) to following insert query with multiple key fields
 func OnConflictUpdateMulti(db *gorm.DB, keys []string, updateCols ...string) *gorm.DB {
-	updateExprs := make([]UpExpr, len(updateCols))
-	for i, v := range updateCols {
-		updateExprs[i] = UpExpr{v, fmt.Sprintf("excluded.%s", v)}
```

Comment:
I don't see the excluded keyword anywhere. How is this handled ?

Reply:
GORMv2 uses Clauses, here Clauses are build, and they are then parsed in batch.go on line 165.

Reply:
I get that, but where does the `excluded.` come from ? I don't see it anywhere

Reply:
It comes from GORM itself. You just specify which columns you want to update. GORM doesn't allow expressions in UPSERT, so I wrote that parser for that.